src/constraints.jl

```diff
@@ -682,15 +684,16 @@ function _moi_add_constraint(
 end
 
 """
-    add_constraint(model::Model, con::AbstractConstraint, name::String="")
+    add_constraint(model::GenericModel, con::AbstractConstraint, name::String="")
 
 Add a constraint `con` to `Model model` and sets its name.
 """
 function add_constraint(
-    model::Model,
+    model::GenericModel,
     con::AbstractConstraint,
     name::String = "",
 )
+    con = model_convert(model, con)
```

Comment:
@blegat: it was sufficient to fix the extension in the docs by adding this here. This means some extra allocations though, so we should probably figure out when we don't need to `model_convert` objects?